### PR TITLE
Feature: Added styling for Cropper

### PR DESCRIPTION
**Pull request by ceesvanegmond**
## Title
Added styling for Cropper feature in https://github.com/wearejust/form-bundle/blob/1.x/Resources/doc/formtypes/image.md

## Changelog

```markdown
### Added
- Added some new CSS specifically for cropper
- Added new JS to make cropper more UI friendly
```

### CODE REVIEW

File: Resources/public/javascript/cropper.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+$(function(){
+    $('.cropper').each(function(index,item){
+        new cropSave($(item));
+    });
+});
+
+
+
```
Comment (Emroni) [inline Resources/public/javascript/cropper.js]: Why use an entire class if the only thing it does it fire **$('*[name="btn_update_and_edit"]').click();**? Just doing this should work too:

$('*[data-method="getCroppedCanvas"]').on('click', function() {
    $('*[name="btn_update_and_edit"]').click();
});

**Review by RutgerKooijman — DISMISSED** [review]
@bartjust Looks good, maybe better to remove empty line ins .cropper class between classnames.

File: Resources/public/javascript/cropper.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+let cropTrigger;
+
+class cropSave {
+    constructor(item) {
+        this.item = item
+        cropTrigger = $('*[data-method="getCroppedCanvas"]');
```
Comment (Emroni) [inline Resources/public/javascript/cropper.js]: You should use **this.item.find()** here

File: Resources/public/javascript/cropper.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+let cropTrigger;
```
Comment (Emroni) [inline Resources/public/javascript/cropper.js]: Move this to the constructor

File: Resources/public/javascript/cropper.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+class cropSave {
```
Comment (Emroni) [inline Resources/public/javascript/cropper.js]: Try to use the same class name as the selector at the bottom, eg: .cropper -> Cropper.

**Review by Emroni — CHANGES_REQUESTED** [review]

**Comment by bartjust** [comment]
@Emroni I've refactored the code. Multiple croppers can be used without problems now.

File: Resources/public/javascript/cropper.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+class Cropper {
+    constructor(item) {
+        this.item = item;
+        this.item.find($('*[data-method="getCroppedCanvas"]')).on('click', this.save.bind(this));
```
Comment (Emroni) [inline Resources/public/javascript/cropper.js]: Almost! 
find( **$(** '*[data-method="getCroppedCanvas"] '**)** )
should be without the **$()**
find('*[data-method="getCroppedCanvas"]')

**Review by Emroni — CHANGES_REQUESTED** [review]

File: Resources/public/javascript/cropper.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    } 
+
+    save() {
+        let updateBtn = $('*[name="btn_update_and_edit"]');
+        if(!updateBtn.length) return;
```
Comment (Emroni) [inline Resources/public/javascript/cropper.js]: So the only function this class has is to save, but why should the class exist in the first place if there is not save button available?

**Comment by bartjust** [comment]
@Emroni The cropper js is removed as the functionality for triggering the save is not preferred anymore. The cropper will now have its default behaviour. This branch only has adjusted styling.

**Pull request merged**